students/khtruong/lesson_09/mailroom_oop.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python
+
+"""This module runs the mailroom script using oop paradigm."""
+
+from datetime import date
+
+
+class Donor:
+    def __init__(self, firstname, lastname, donations=None):
+        self.firstname = firstname
+        self.lastname = lastname
+        self.fullname = '{} {}'.format(firstname, lastname)
+        if donations:
+            self.donations = donations
+        else:
+            self.donations = []
+
+    def __lt__(self, other):
+        return ((self.totaldonation, self.firstname) <
+                (other.totaldonation, other.firstname))
+
+    def add_donation(self, amount):
+        self.donations.append(amount)
+
+    @property
+    def totaldonation(self):
+        return sum(self.donations)
+
+    @property
+    def numdonation(self):
+        return len(self.donations)
+
+    @property
+    def avgdonation(self):
+        return self.totaldonation / max(self.numdonation, 1)
+
+    @classmethod
+    def from_fullname(cls, fullname, donations):
+        firstname, lastname = fullname.split(' ')
+        return cls(firstname, lastname, donations)
+
+
+class DataBase:
+
+    def __init__(self, donors=None):
+        if donors:
+            self.donors = donors
+        else:
+            self.donors = []
+
+    def donor_and_amount(self):
+        """Return prompt for donor name and donation amount and print to
+        screen.
+        """
+        fullname = fullname_input()
```

Comment:
"There should be no input function in the class that holds the data!"

as mentioned in the instructions you should separate the user inputs from your data handling code.

the user interaction code should be calling this class with already collected input like: 
`dbobj.add_donation(amount)` where amount was collected outside of class

doing this keeps the class clean and on topic plus easier to unit test the class itself

Reply:
to give you a more practical example, imagine you are now asked to write a GUI for this mailroom program instead of cli.

you should be able to take your `Database` class and use it in the front end code without refactor